File: tmp-find-ambiguous-joins.py
import pathlib, re
root = pathlib.Path('g:/centro-cultural')
file_patterns = ['*.ts', '*.tsx', '*.js', '*.jsx']
ignore_dirs = {'node_modules', '.next', 'out', 'dist', 'public', 'supabase'}

select_re = re.compile(r"\.select\(\s*([`'])(.*?)\1", re.S)
rel_re = re.compile(r"^\s*([a-z0-9_]+)\s*(?:[:][a-z0-9_]+)?\s*\(", re.I)

for path in root.rglob('*'):
    if path.is_dir():
        continue
    if any(path.match(pat) for pat in file_patterns):
        if any(part in ignore_dirs for part in path.parts):
            continue
        text = path.read_text(encoding='utf-8', errors='ignore')
        for m in select_re.finditer(text):
            arg = m.group(2)
            if '!' in arg:
                continue
            if '(' not in arg:
                continue
            lines = arg.splitlines()
            rels = [line.strip() for line in lines if rel_re.match(line.strip())]
            if rels:
                print(f'{path}:{m.start()}')
                for line in rels:
                    print('  ', line)
                print()

File: tmp-find-ambiguous-joins.js
const fs = require('fs');
const path = require('path');
const root = path.resolve('g:/centro-cultural');
const filePatterns = ['.ts', '.tsx', '.js', '.jsx'];
const ignoreDirs = new Set(['node_modules', '.next', 'out', 'dist', 'public', 'supabase']);
const relRe = /^\s*([a-z0-9_]+)\s*(?:[:][a-z0-9_]+)?\s*\(/i;

function walk(dir) {
  const entries = fs.readdirSync(dir, { withFileTypes: true });
  for (const entry of entries) {
    if (ignoreDirs.has(entry.name)) continue;
    const fullPath = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      walk(fullPath);
      continue;
    }
    if (!filePatterns.includes(path.extname(entry.name).toLowerCase())) continue;
    const text = fs.readFileSync(fullPath, 'utf8');
    const regex = /\.select\(\s*([`'])([\s\S]*?)\1/g;
    let match;
    while ((match = regex.exec(text))) {
      const arg = match[2];
      if (!arg.includes('(') || arg.includes('!')) continue;
      const lines = arg.split(/\r?\n/);
      const rels = lines.filter((line) => relRe.test(line.trim()));
      if (rels.length > 0) {
        console.log(`${fullPath}:${match.index}`);
        rels.forEach((line) => console.log('  ', line.trim()));
        console.log('');
      }
    }
  }
}

walk(root);
